### PR TITLE
Remove buffer connected to single port

### PR DESCRIPTION
**Pull request by mathursanjiv**

### CODE REVIEW

File: src/rsz/test/remove_buffers1.ok [hunk 1]
```diff
@@ -39,12 +39,13 @@ Path Type: max
...
-           0.05   data arrival time
+   0.02    0.02 v b4/Z (BUF_X8)
+   0.00    0.02 v out1 (out)
+           0.02   data arrival time
 ---------------------------------------------------------
 (Path is unconstrained)
 
 
-No differences found.
+Differences found at line 7.
```
Comment (jjcherry56) [inline src/rsz/test/remove_buffers1.ok:49]: you did not use save_defok to update the defok file

File: src/rsz/test/remove_buffers2.ok [hunk 2]
```diff
@@ -26,20 +26,22 @@ Path Type: max
...
+   0.00    0.00 v b2/A (BUF_X4)
+   0.02    0.02 v b2/Z (BUF_X4)
+   0.00    0.02 v out2 (out)
            0.02   data arrival time
 ---------------------------------------------------------
 (Path is unconstrained)
 
 
-No differences found.
+Differences found at line 7.
```
Comment (jjcherry56) [inline src/rsz/test/remove_buffers2.ok:45]: again

File: src/rsz/src/Resizer.cc [hunk 3]
```diff
@@ -321,7 +321,7 @@ Resizer::bufferConnectedToPorts(Instance *buffer)
   Net *out_net = db_network_->net(out_pin);
```
Comment (jjcherry56) [inline src/rsz/src/Resizer.cc:321]: bufferConnectedToPorts is no longer an appropriate function name.
bufferBetweenPorts is more appropriate.

The comment in the caller needs to be updated in the same way.

**Comment by jjcherry56** [comment]
Then you obviously can't check in the changes without resolving the problems they cause

**Comment by mathursanjiv** [comment]
Removing only buffers connected to clock ports since only these give problem in clock building.
Now regression failures are small.

Flow
gcd_nangate45 pass
gcd_sky130hs pass
gcd_sky130hd pass
ibex_sky130hd pass
ibex_sky130hs pass
aes_nangate45 pass
aes_sky130hs pass
aes_sky130hd pass
tinyRocket_nangate45 pass
jpeg_sky130hd *FAIL* max_slew_violations        4 >        2
jpeg_sky130hs *FAIL* clock_skew    0.638 >    0.595

File: src/rsz/src/Resizer.cc [hunk 4]
```diff
@@ -331,8 +331,9 @@ Resizer::bufferConnectedToPorts(Instance *buffer)
   Net *in_net = db_network_->net(in_pin);
   Net *out_net = db_network_->net(out_pin);
   bool in_net_ports = hasPort(in_net);
+  bool in_net_clk = sta_->isClock(in_net);
   bool out_net_ports = hasPort(out_net);
-  return in_net_ports || out_net_ports;
+  return (in_net_ports && !in_net_clk) || out_net_ports;
```
Comment (jjcherry56) [inline src/rsz/src/Resizer.cc:336]: once again you have changed the function and left the function name inconsistent.

Did you check the failures with just the merge and not this change?
I think some of the recent resizer changes may be relevant.

Comment (mathursanjiv) [reply src/rsz/src/Resizer.cc:336]: Yes, tests pass after merge. I have changed function name back.

**Comment by mathursanjiv** [comment]
@jjcherry56 if you are ok, can you approve this PR? Thanks.

**Comment by jjcherry56** [comment]
I think you have regressed. There is no reason to special case the clock net.
 Change function name from bufferConnectedToPorts to bufferBetweenPorts commit 4914bee20821353a28c6cee965f69561e824e8f3 was the correct solution, you just need to update the flow metrics (which probably pass now because of changes since then).

**Comment by mathursanjiv** [comment]
> I think you have regressed. There is no reason to special case the clock net.
> Change function name from bufferConnectedToPorts to bufferBetweenPorts commit [4914bee](https://github.com/The-OpenROAD-Project/OpenROAD/commit/4914bee20821353a28c6cee965f69561e824e8f3) was the correct solution, you just need to update the flow metrics (which probably pass now because of changes since then).

Done, please check now for approval.

**Comment by jjcherry56** [comment]
you seem to have removed the whole reason for the PR, which is changing the || to && in bufferConnectedToPorts

**Comment by mathursanjiv** [comment]
> you seem to have removed the whole reason for the PR, which is changing the || to && in bufferConnectedToPorts

Yes, got into confusion! Quite a few flow regressions files have been updated to take care of failures.

**Review by jjcherry56 — APPROVED** [review]

**Pull request merged**